trusted.ci.jenkins.io.tf

```
@@ -105,6 +105,17 @@ module "trusted_ci_jenkins_io_azurevm_agents" {
   }
 }
 
+# Required to allow azcopy sync of updates.jenkins.io File Share with the permanent agent
+module "trusted_ci_jenkins_io_storage" {
+  source = "./.shared-tools/terraform/modules/azure-jenkinsinfra-storage"
+
+  service_fqdn                  = module.trusted_ci_jenkins_io.service_fqdn
+  storage_service_principal_id  = module.trusted_ci_jenkins_io.controler_service_principal_id
```

```suggestion
  storage_service_principal_ids  = [
    data.azuread_service_principal.terraform_production.id,
  ]
```

Same as https://github.com/jenkins-infra/azure/blob/main/ci.jenkins.io.tf#L39-L41:

- This variable tells the module who is (are) the owner(s) of the application. We usually specify the ID of the terraform's SP used to execute terraform (or the account admins IDs  in special cases)
- Using the existing controller SP is misleading as it would allow it to use the SP at system level to write to the storage accounts while we only want agents to do it

I think the confusion comes from the poor naming I used in the `azure-jenkinsinfra-controller` module: 
https://github.com/jenkins-infra/shared-tools/blob/main/terraform/modules/azure-jenkinsinfra-controller/variables.tf#L38-L40

---

(see https://github.com/jenkins-infra/shared-tools/pull/131#pullrequestreview-1789390674)